Document `drop_extended_stats` procedure and `analyze` statement in Iceberg (#14490)
trinodb/trino: docs/src/main/sphinx/connector/iceberg.rst

``Retention specified (1.00d) is shorter than the minimum retention configured in the system (7.00d)``.
The default value for this property is ``7d``.

drop_extended_stats
~~~~~~~~~~~~~~~~~~~

> The ``drop_extended_stats`` command removes extended statistics from metadata file.

Reviewer: from the metadata file. 

or 

from metadata files.

Reviewer: also.. do we need to explain what extended stats includes. Or link to relevant info?

``Retention specified (1.00d) is shorter than the minimum retention configured in the system (7.00d)``.
The default value for this property is ``7d``.

.. _drop-extended-stats:

drop_extended_stats

> ~~~~~~~~~~~~~~~~~~~

Reviewer 2: let's have a note this is experimental

> The ``drop_extended_stats`` command removes extended statistics from metadata files.

Reviewer 2: > from metadata files.

from the table

``drop_extended_stats`` can be run as follows:

.. code-block:: sql

>   ALTER TABLE test_table EXECUTE drop_extended_stats

Reviewer 2: ```suggestion
  SET SESSION my_catalog.experimental_extended_statistics_enabled = true;
  ALTER TABLE test_table EXECUTE drop_extended_stats
```

Author: I believe `drop_extended_stats` isn't protected by the session property. Is it a bug? 
```sql
trino> SHOW SESSION LIKE 'iceberg.experimental_extended_statistics_enabled';
                       Name                       | Value | Default |  Type   |                                                          Description
--------------------------------------------------+-------+---------+---------+--------------------------------------------------------------------------------------------------------------------------------
 iceberg.experimental_extended_statistics_enabled | false | false   | boolean | Allow ANALYZE and use of extended statistics collected by it. Currently, the statistics are collected in Trino-specific format

trino> ALTER TABLE iceberg.tpch.region EXECUTE drop_extended_stats;
ALTER TABLE EXECUTE
 rows
------
(0 rows)
```

Reviewer 2: You're right, it's not.

It doesn't have to be -- it's definitely important to protect the ANALYZE and SELECT code paths, but this one is just helper, but can become protected as well.

Dropping a materialized view with :doc:`/sql/drop-materialized-view` removes
the definition and the storage table.

Table statistics
----------------

There is experimental support to collect column statistics which can be enabled by
setting the ``iceberg.experimental.extended-statistics.enabled`` catalog
configuration property or the corresponding
``experimental_extended_statistics_enabled`` session property to ``true``.

> Enabling this configuration will allow executing :doc:`/sql/analyze` statement.

Reviewer: ```suggestion
Enabling this configuration allow executing :doc:`/sql/analyze` statement to gather statistics.
```

.. _iceberg_analyze:

Updating table statistics
^^^^^^^^^^^^^^^^^^^^^^^^^

If your queries are complex and include joining large data sets,

> running :doc:`/sql/analyze` on tables may improve query performance

Reviewer: why is wrapping narrower here?

Author: What do you mean?

Author: Are we following line wrapping limitation strictly? I think no.

by collecting statistical information about the data::

    ANALYZE table_name

> This query will collect statistics for all columns.

Reviewer: ```suggestion
This query collects statistics for all columns.
```

On wide tables, collecting statistics for all columns can be expensive and can have a
detrimental effect on query planning. It is also typically unnecessary - statistics are

> only useful on specific columns, like join keys, predicates, grouping keys. One can

Reviewer: You can

> specify a subset of columns to be analyzed via the optional ``columns`` property::

Reviewer: to analyze with the

    ANALYZE table_name WITH (columns = ARRAY['col_1', 'col_2'])

This query collects statistics for columns ``col_1`` and ``col_2``.

Note that if statistics were previously collected for all columns, they need to be dropped

> before re-analyzing just a subset::

Reviewer: Dont repeat the info .. link to the other section

Dropping a materialized view with :doc:`/sql/drop-materialized-view` removes
the definition and the storage table.

Table statistics
----------------

There is experimental support to collect column statistics which can be enabled by
setting the ``iceberg.experimental.extended-statistics.enabled`` catalog
configuration property or the corresponding
``experimental_extended_statistics_enabled`` session property to ``true``.
Enabling this configuration allows executing :doc:`/sql/analyze` statement to gather statistics.

> 

Reviewer: We have to add if it is necessary to run analyse or if stats are also collected and/or update on the fly with data changes (insert, update, delete..)

.. _iceberg_analyze:

Updating table statistics
^^^^^^^^^^^^^^^^^^^^^^^^^

If your queries are complex and include joining large data sets,
running :doc:`/sql/analyze` on tables may improve query performance
by collecting statistical information about the data::

    ANALYZE table_name

This query collects statistics for all columns.

On wide tables, collecting statistics for all columns can be expensive and can have a

> detrimental effect on query planning. It is also typically unnecessary - statistics are

Reviewer 2: > detrimental effect on query planning

remove. i don't think it's currently true for Iceberg

> only useful on specific columns, like join keys, predicates, grouping keys. You can

Reviewer 3: ```suggestion
only useful on specific columns, like join keys, predicates, or grouping keys. You can
```
